### PR TITLE
Add a template hook `regformBeforeSections` to the regform

### PR DESCRIPTION
**Pull request by vasantvohra**
Hello, At the UN plugin, we have a representation type section that should remain on the top, and it's not customizable by the users like the vehicle access plugin. Overall so far 80% of implementation, I just need this template hook upstream.

### CODE REVIEW

File: indico/modules/events/registration/client/js/form_submission/RegistrationFormSubmission.jsx [hunk 1]
```diff
@@ -175,6 +175,7 @@ export default function RegistrationFormSubmission() {
     >
       {fprops => (
         <form onSubmit={fprops.handleSubmit}>
+          {renderPluginComponents('regformBeforeSections')}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/client/js/form_submission/RegistrationFormSubmission.jsx:178]: Oh great, looks like we have a mix of  camelCase and snake-case for this...

Comment (vasantvohra) [reply indico/modules/events/registration/client/js/form_submission/RegistrationFormSubmission.jsx:178]: similar to
` {renderPluginComponents('regformAfterSections')}`
- https://github.com/indico/indico/pull/6218/files#diff-531ba68586d2b7bbb75806adb77fcc16170e1a61fbeaa65b37924b0e947b7941L181

Comment (ThiefMaster) [reply indico/modules/events/registration/client/js/form_submission/RegistrationFormSubmission.jsx:178]: yeah, i think i used camelcase in the very first ones i added, and then some others were added as snake-case and i didn't realize we had camelCase here...

Comment (vasantvohra) [reply indico/modules/events/registration/client/js/form_submission/RegistrationFormSubmission.jsx:178]: Thanks for the prompt merge :)

**Pull request merged**